packages/recoil/recoil_values/Recoil_atom.js

```diff
@@ -292,6 +293,23 @@ function baseAtom<T>(options: BaseAtomOptions<T>): RecoilState<T> {
         return getLoadable(recoilValue).toPromise();
       }
 
+      function observe<S>(
+        recoilValue: RecoilValue<S>,
+        observer: (value: S) => void,
```

Comment:
Could potentially also get the previous value and `isReset` consistent with `onSet()`.  But, that could also be added later, if needed.

Reply:
Yep, as pointed out above I think we'll need access to the defaultLoadable on the upstream RecoilValue, which may require some refactoring.